metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ExecutionAggregator.java

```diff
@@ -1,56 +1,115 @@
+
+    /**
+     * Get the executions containing the metric to aggregate from ExecutionsRequestBody.
+     * @param executionsRequestBody
+     * @return
+     */
+    public abstract List<T> getExecutionsFromExecutionRequestBody(ExecutionsRequestBody executionsRequestBody);
+
+    /**
+     * Returns a boolean indicating if the execution metric is valid.
+     * @param executionMetric
+     * @return
+     */
+    public abstract boolean validateExecutionMetric(E executionMetric);
 
     /**
      * Aggregates workflow executions into an aggregated metric.
-     * @param executions
+     * @param executionMetrics
      * @return
      */
-    Optional<M> getAggregatedMetricFromExecutions(List<T> executions);
+    protected abstract Optional<M> calculateAggregatedMetricFromExecutionMetrics(List<E> executionMetrics);
 
     /**
      * Aggregates a list of aggregated metrics into one aggregated metric.
      * @param aggregatedMetrics
      * @return
      */
-    Optional<M> getAggregatedMetricsFromAggregatedMetrics(List<M> aggregatedMetrics);
+    protected abstract Optional<M> calculateAggregatedMetricFromAggregatedMetrics(List<M> aggregatedMetrics);
+
+    /**
+     * Aggregate metrics from all submissions in the ExecutionsRequestBody.
+     * @param allSubmissions
+     * @return
+     */
+    public abstract Optional<M> getAggregatedMetricFromAllSubmissions(ExecutionsRequestBody allSubmissions);
 
     /**
-     * Returns a list of executions of type T where the execution metric is not null.
+     * Aggregates workflow executions into an aggregated metric and calculates the number of skipped executions.
      * @param executions
      * @return
      */
-    default List<E> getNonNullMetricsFromExecutions(List<T> executions) {
-        return executions.stream()
-                .map(this::getMetricFromExecution)
+
+    public final Optional<M> getAggregatedMetricFromExecutions(List<T> executions) {
+        final List<E> executionsWithNonNullMetric = executions.stream().map(this::getMetricFromExecution).filter(Objects::nonNull).toList();
+        final List<E> validExecutionMetrics = executionsWithNonNullMetric.stream().filter(this::validateExecutionMetric).toList();
+        if (!validExecutionMetrics.isEmpty()) {
+            Optional<M> calculatedMetric = calculateAggregatedMetricFromExecutionMetrics(validExecutionMetrics);
+            final int numberOfSkippedExecutions = executionsWithNonNullMetric.size() - validExecutionMetrics.size();
+            calculatedMetric.ifPresent(metric -> metric.setNumberOfSkippedExecutions(numberOfSkippedExecutions));
+            return calculatedMetric;
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Aggregates a list of aggregated metrics into one aggregated metric  nd calculates the number of skipped executions.
+     * @param aggregatedMetrics
+     * @return
+     */
+    public final Optional<M> getAggregatedMetricsFromAggregatedMetrics(List<M> aggregatedMetrics) {
```

Comment:
Method that calculates the aggregated metric plus sets the number of executions skipped.

Reply:
Since it's calculating a single aggregated metric, should it be named `getAggregatedMetricFromAggregatedMetrics`?

Reply:
There is an instance where it calculates multiple aggregated metrics into a single aggregated metric: https://github.com/dockstore/dockstore-support/blob/64ddd193e99795c9dca1f0f27af5f400110bf05f/metricsaggregator/src/main/java/io/dockstore/metricsaggregator/MetricsAggregatorS3Client.java#L131-L133